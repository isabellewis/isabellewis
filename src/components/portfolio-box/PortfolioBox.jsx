import React, { PureComponent } from 'react';
import styled from 'styled-components';
// import img from '../../assets/portfolio/london2.jpg';

const PortfolioBoxWrapper = styled.div`
  display: none;

	@media only screen and (min-width: 576px) {
    display: block;
    height: 50%;
    padding: 60px 0;
    text-align: center;
    width: 100%;
	}
`;

// background: url(${img});
const PortfolioBoxContent = styled.div`
  background-repeat: no-repeat;
  background-size: cover;
  display: inline-block;
  height: 100%;
  opacity: .5;
  overflow: hidden;
  width: 50%;
`;

const ArrowIcon = styled.i`
  color: #ffaeae;
  cursor: pointer;
  display: inline-block;
  font-size: 100px;
  user-select: none;

	@media only screen and (min-width: 768px) {
		font-size: 150px;
  }
  
	@media only screen and (min-width: 992px) {
		font-size: 200px;
	}
`;

const PortfolioBoxTitle = styled.h2`
  color: #000;
  font-family: 'Archivo Narrow',sans-serif;
  font-size: 45px;
  opacity: 1;
`;

const PortfolioItemDiv = styled.div`
  background: #ffaeae;
  display: inline-block;
  height: 250px;
  margin: 0 30px 0 0;
  overflow-wrap: break-word;
  padding: 30px;
  vertical-align: middle;
  white-space: normal;
  width: 250px;
`;

const UpperPanel = styled.div`
  height: 100%;
`;

const LowerPanel = styled.div`
  padding: 50px 0;
`;

export class PortfolioBox extends PureComponent {
  constructor(props) {
		super(props);

		this.state = {
			activeIndex: 0
    }
    
    this.arrowDecrease = this.arrowDecrease.bind(this);
    this.arrowIncrease = this.arrowIncrease.bind(this);
  }
  
  arrowDecrease() {
    const {
      projectData
    } = this.props;

    const {
      activeIndex
    } = this.state;

    const newIndex = activeIndex > 0 ? activeIndex - 1 : projectData.projects.length -1;

    this.setState (
      {
        activeIndex: newIndex
      }
    )
  }

  arrowIncrease() {
    const {
      projectData
    } = this.props;

    const {
      activeIndex
    } = this.state;

    const newIndex = activeIndex < projectData.projects.length - 1 ? activeIndex + 1 : 0;

    this.setState (
      {
        activeIndex: newIndex
      }
    )
  }

  render() {
    const {
      projectData
    } = this.props;

    const {
      activeIndex
    } = this.state;


    return (
      <PortfolioBoxWrapper>
        <UpperPanel>
          <ArrowIcon className="material-icons" onClick={this.arrowDecrease}>{this.state.menuOpen ? 'close' : '' }keyboard_arrow_left</ArrowIcon>
          <PortfolioBoxContent>
            <PortfolioBoxTitle>{this.state.activeIndex}</PortfolioBoxTitle>
          </PortfolioBoxContent>
          <ArrowIcon className="material-icons" onClick={this.arrowIncrease}>keyboard_arrow_right</ArrowIcon>
        </UpperPanel>
        <LowerPanel>
          <PortfolioItemDiv>
            <h2>{projectData.projects[activeIndex > 0 ? activeIndex - 1 : projectData.projects.length - 1].name}</h2>
          </PortfolioItemDiv>
          <PortfolioItemDiv>
            <h2>{projectData.projects[activeIndex].name}</h2>
          </PortfolioItemDiv>
          <PortfolioItemDiv>
            <h2>{projectData.projects[activeIndex < projectData.projects.length -1 ? activeIndex + 1 : 0].name}</h2>
          </PortfolioItemDiv>
        </LowerPanel>
      </PortfolioBoxWrapper>
    );
  }
};
