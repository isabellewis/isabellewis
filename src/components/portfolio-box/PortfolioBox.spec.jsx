import React from 'react';
import { shallow } from 'enzyme';
import { PortfolioBox } from './PortfolioBox';

const projectData = {
  projects: [
    {
      name: '0React Project Name Goes Here',
      technologies: 'HTML, CSS, JavaScript, Webpack',
      background: '#ffaeae',
      github: 'https://github.com/isabellewis/isabellewis',
      livewebsite: 'https://github.com/isabellewis/isabellewis',
      description: 'Descriptions will go here'
    },
    {
      name: '1React Project Name Goes Here',
      technologies: 'HTML, CSS, JavaScript, Webpack',
      background: '#ffaeae',
      github: 'https://github.com/isabellewis/isabellewis',
      livewebsite: 'https://github.com/isabellewis/isabellewis',
      description: 'Descriptions will go here'
    },
    {
      name: '2React Project Name Goes Here',
      technologies: 'HTML, CSS, JavaScript, Webpack',
      background: '#ffaeae',
      github: 'https://github.com/isabellewis/isabellewis',
      livewebsite: 'https://github.com/isabellewis/isabellewis',
      description: 'Descriptions will go here'
    },
    {
      name: '3React Project Name Goes Here',
      technologies: 'HTML, CSS, JavaScript, Webpack',
      background: '#ffaeae',
      github: 'https://github.com/isabellewis/isabellewis',
      livewebsite: 'https://github.com/isabellewis/isabellewis',
      description: 'Descriptions will go here'
    },
    {
      name: '4React Project Name Goes Here',
      technologies: 'HTML, CSS, JavaScript, Webpack',
      background: '#ffaeae',
      github: 'https://github.com/isabellewis/isabellewis',
      livewebsite: 'https://github.com/isabellewis/isabellewis',
      description: 'Descriptions will go here'
    }
  ]
}


describe('PortfolioBox', () => {
	let component;

	beforeEach(() => {
		component = shallow(<PortfolioBox projectData={projectData} />);
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});