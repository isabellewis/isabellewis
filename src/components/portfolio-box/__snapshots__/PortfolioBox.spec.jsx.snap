// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioBox should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PortfolioBox
    projectData={
      Object {
        "projects": Array [
          Object {
            "background": "#ffaeae",
            "description": "Descriptions will go here",
            "github": "https://github.com/isabellewis/isabellewis",
            "livewebsite": "https://github.com/isabellewis/isabellewis",
            "name": "0React Project Name Goes Here",
            "technologies": "HTML, CSS, JavaScript, Webpack",
          },
          Object {
            "background": "#ffaeae",
            "description": "Descriptions will go here",
            "github": "https://github.com/isabellewis/isabellewis",
            "livewebsite": "https://github.com/isabellewis/isabellewis",
            "name": "1React Project Name Goes Here",
            "technologies": "HTML, CSS, JavaScript, Webpack",
          },
          Object {
            "background": "#ffaeae",
            "description": "Descriptions will go here",
            "github": "https://github.com/isabellewis/isabellewis",
            "livewebsite": "https://github.com/isabellewis/isabellewis",
            "name": "2React Project Name Goes Here",
            "technologies": "HTML, CSS, JavaScript, Webpack",
          },
          Object {
            "background": "#ffaeae",
            "description": "Descriptions will go here",
            "github": "https://github.com/isabellewis/isabellewis",
            "livewebsite": "https://github.com/isabellewis/isabellewis",
            "name": "3React Project Name Goes Here",
            "technologies": "HTML, CSS, JavaScript, Webpack",
          },
          Object {
            "background": "#ffaeae",
            "description": "Descriptions will go here",
            "github": "https://github.com/isabellewis/isabellewis",
            "livewebsite": "https://github.com/isabellewis/isabellewis",
            "name": "4React Project Name Goes Here",
            "technologies": "HTML, CSS, JavaScript, Webpack",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.i
            className="material-icons"
            onClick={[Function]}
          >
            
            keyboard_arrow_left
          </styled.i>
          <styled.div>
            <styled.h2>
              0
            </styled.h2>
          </styled.div>
          <styled.i
            className="material-icons"
            onClick={[Function]}
          >
            keyboard_arrow_right
          </styled.i>
        </styled.div>,
        <styled.div>
          <styled.div>
            <h2>
              4React Project Name Goes Here
            </h2>
          </styled.div>
          <styled.div>
            <h2>
              0React Project Name Goes Here
            </h2>
          </styled.div>
          <styled.div>
            <h2>
              1React Project Name Goes Here
            </h2>
          </styled.div>
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.i
              className="material-icons"
              onClick={[Function]}
            >
              
              keyboard_arrow_left
            </styled.i>,
            <styled.div>
              <styled.h2>
                0
              </styled.h2>
            </styled.div>,
            <styled.i
              className="material-icons"
              onClick={[Function]}
            >
              keyboard_arrow_right
            </styled.i>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "",
                "keyboard_arrow_left",
              ],
              "className": "material-icons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "",
              "keyboard_arrow_left",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.h2>
                0
              </styled.h2>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "keyboard_arrow_right",
              "className": "material-icons",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "keyboard_arrow_right",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <h2>
                4React Project Name Goes Here
              </h2>
            </styled.div>,
            <styled.div>
              <h2>
                0React Project Name Goes Here
              </h2>
            </styled.div>,
            <styled.div>
              <h2>
                1React Project Name Goes Here
              </h2>
            </styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h2>
                4React Project Name Goes Here
              </h2>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "4React Project Name Goes Here",
              },
              "ref": null,
              "rendered": "4React Project Name Goes Here",
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h2>
                0React Project Name Goes Here
              </h2>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "0React Project Name Goes Here",
              },
              "ref": null,
              "rendered": "0React Project Name Goes Here",
              "type": "h2",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <h2>
                1React Project Name Goes Here
              </h2>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "1React Project Name Goes Here",
              },
              "ref": null,
              "rendered": "1React Project Name Goes Here",
              "type": "h2",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.i
              className="material-icons"
              onClick={[Function]}
            >
              
              keyboard_arrow_left
            </styled.i>
            <styled.div>
              <styled.h2>
                0
              </styled.h2>
            </styled.div>
            <styled.i
              className="material-icons"
              onClick={[Function]}
            >
              keyboard_arrow_right
            </styled.i>
          </styled.div>,
          <styled.div>
            <styled.div>
              <h2>
                4React Project Name Goes Here
              </h2>
            </styled.div>
            <styled.div>
              <h2>
                0React Project Name Goes Here
              </h2>
            </styled.div>
            <styled.div>
              <h2>
                1React Project Name Goes Here
              </h2>
            </styled.div>
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.i
                className="material-icons"
                onClick={[Function]}
              >
                
                keyboard_arrow_left
              </styled.i>,
              <styled.div>
                <styled.h2>
                  0
                </styled.h2>
              </styled.div>,
              <styled.i
                className="material-icons"
                onClick={[Function]}
              >
                keyboard_arrow_right
              </styled.i>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "",
                  "keyboard_arrow_left",
                ],
                "className": "material-icons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "",
                "keyboard_arrow_left",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <styled.h2>
                  0
                </styled.h2>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": 0,
                },
                "ref": null,
                "rendered": 0,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "keyboard_arrow_right",
                "className": "material-icons",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "keyboard_arrow_right",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <h2>
                  4React Project Name Goes Here
                </h2>
              </styled.div>,
              <styled.div>
                <h2>
                  0React Project Name Goes Here
                </h2>
              </styled.div>,
              <styled.div>
                <h2>
                  1React Project Name Goes Here
                </h2>
              </styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h2>
                  4React Project Name Goes Here
                </h2>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "4React Project Name Goes Here",
                },
                "ref": null,
                "rendered": "4React Project Name Goes Here",
                "type": "h2",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h2>
                  0React Project Name Goes Here
                </h2>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0React Project Name Goes Here",
                },
                "ref": null,
                "rendered": "0React Project Name Goes Here",
                "type": "h2",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <h2>
                  1React Project Name Goes Here
                </h2>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1React Project Name Goes Here",
                },
                "ref": null,
                "rendered": "1React Project Name Goes Here",
                "type": "h2",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
